bundles/binding/org.openhab.binding.caldav-command/src/main/java/org/openhab/binding/caldav_command/internal/CalDavBinding.java

```diff
@@ -0,0 +1,359 @@
+				if (disabledItems.contains(eventContent.getItem().getName())) {
+					continue;
+				}
+				
+				EventUtils.EventContent currentEventContent = map.get(eventContent.getItem().getName());
+				if (eventContent.getTime().isBefore(DateTime.now())
+						&& (currentEventContent == null || eventContent.getTime().isAfter(currentEventContent.getTime()))) {
+					map.put(eventContent.getItem().getName(), eventContent);
+				}
+			}
+		}
+		
+		for (EventUtils.EventContent currentEventContent : map.values()) {
+			eventPublisher.sendCommand(currentEventContent.getItem().getName(), currentEventContent.getCommand());
+			logger.debug("setting initial value for {} to {}", currentEventContent.getItem().getName(), currentEventContent.getCommand());
+		}
+	}
+	
+	private void doAction(CalDavEvent event, String scope) {
+		final List<EventUtils.EventContent> parseContent = EventUtils.parseContent(event, this.itemRegistry, scope);
+		for (EventUtils.EventContent eventContent : parseContent) {
+			if (disabledItems.contains(eventContent.getItem().getName())) {
+				continue;
+			}
+			
+			eventPublisher.sendCommand(eventContent.getItem().getName(), eventContent.getCommand());
+		}
+	}
+	
+	private void updateItemState(CalDavNextEventConfig config, List<CalDavEvent> events) {
+		String itemName = config.getItemNameToListenTo();
+		String itemNamePreview = config.getItemName();
+		logger.trace("update item state for item: {}", itemName);
+		
+		Command state = null;
+		DateTime time = null;
+		
+		if (calDavLoader == null) {
+			logger.warn("caldav loader is not set");
+			return;
+		}
+		
+		for (CalDavEvent calDavEvent : events) {
+			Item item = null;
+			try {
+				item = this.itemRegistry.getItem(itemName);
+			} catch (ItemNotFoundException e) {
+				logger.error("item {} could not be found", itemName);
+				continue;
+			}
+			
+			final List<EventUtils.EventContent> parseContent = EventUtils.parseContent(calDavEvent, item);
+			for (EventUtils.EventContent eventContent : parseContent) {
+				if (eventContent.getTime().isBefore(DateTime.now())) {
+					continue;
+				}
+				
+				if (time == null || time.isAfter(eventContent.getTime())) {
+					time = eventContent.getTime();
+					state = eventContent.getCommand();
+				}
+			}
+		}
+		
+		if (time == null) {
+			// no item found
+			eventPublisher.postUpdate(itemNamePreview, org.openhab.core.types.UnDefType.UNDEF);
+			return;
+		}
+		
+		CalDavType type = config.getType();
+		logger.trace("handling event of type: {}", type);
+		if (type == CalDavType.VALUE) {
+			logger.debug("setting value for '{}' to: {}", itemNamePreview, state);
+			eventPublisher.sendCommand(itemNamePreview, state);
+		} else if (type == CalDavType.DATE) {
+			Command c = new DateTimeType(FORMATTER.print(time));
+			logger.debug("setting value for '{}' to: {}", itemNamePreview, c);
+			eventPublisher.sendCommand(itemNamePreview, c);
+		} else if (type == CalDavType.DISABLE) {
+			// ok
```

Comment:
If nothing needs to be handled on CalDavType.DISABLE, you could "return" to avoid the empty if statement.